kiosk/src/Components/AddingGame.tsx

```diff
@@ -0,0 +1,161 @@
+            const getGameCodeUrl = `https://staging.pxt.io/api/kiosk/code/${kioskCode}`;
+            const getGameCode = async () => {
+                if (kiosk.state !== KioskState.AddingGame) {
+                    return;
+                }
+                let response = await fetch(getGameCodeUrl);
+                if (!response.ok) {
+                    throw new Error("Unable to get data from the kiosk.");
+                } else {
+                    const gameCode = (await response.json())?.code;
+                    if (gameCode !== "0") {
+                        kiosk.saveNewGame(gameCode);
+                        // this is something we might want to do in the future
+                        // ie if the game gets added, launch it immediately
+                        // instead of going back to the main menu to find it
+                        // kiosk.launchGame(gameCode);
+                        return true;
+                    }
+                    throw new Error("Invalid game code");
+                }
+            }
+            return new Promise<void>((resolve, reject) => {
+                let pollInterval: any;
+                let pollTimeout: any;
+                pollInterval = setInterval(() => {
+                    getGameCode()
+                        .then(() => {
+                            clearInterval(pollInterval);
+                            clearTimeout(pollTimeout);
+                            resolve();
+                        })
+                        .catch(() => null);
+                }, whenToPull);
+                pollTimeout = setTimeout(() => {
+                    clearInterval(pollInterval);
+                    reject();
+                }, timeoutDuration)
+            });
+        }
+    }, [renderQRCode]);
+
+    const qrDivContent = () => {
+        if (renderQRCode) {
+            return (
+                <div className="innerQRCodeContent">
+                    <QRCodeSVG value={`${kioskCodeUrl}#add-game:${kioskCode}`} />
+                    <h3>10 minute Kiosk ID</h3>
+                    <h1>{kioskCode}</h1>
+                </div>
+            )
+        }
+        else {
+            return (
+                <div className="innerQRCodeContent">
+                    <AddGameButton kiosk={kiosk} selected={qrCodeButtonSelected} content="Generate new QR code" />
+                </div>
+            )
+        }
+    };
+
+    return (
+        <div className="addGame">
+            <h1>Add your game</h1>
+            <div className="addGameContent">
+                <div className="addInstructions">
+                    <h2>How to upload your game</h2>
+                    <ol>
+                        <li>Go to your game on your personal phone or computer</li>
+                        <li>Click on the share button</li>
+                        <li>Select "Share to Kiosk"</li>
+                        <li>Add kiosk ID.</li>
```

Comment:
```suggestion
                        <li>Add kiosk ID</li>
```

nit: feels off if 1/4 list elements has punctaitno